libnetwork/netavark/config.go

```diff
@@ -38,6 +38,16 @@ func (n *netavarkNetwork) NetworkCreate(net types.Network) (types.Network, error
 }
 
 func (n *netavarkNetwork) networkCreate(newNetwork *types.Network, defaultNet bool) (*types.Network, error) {
+	if len(newNetwork.NetworkDNSServers) > 0 && !newNetwork.DNSEnabled {
+		return nil, fmt.Errorf("Cannot set NetworkDNSServers if DNS is not enabled for the network: %w", types.ErrInvalidArg)
+	} else if len(newNetwork.NetworkDNSServers) > 0 {
```

Comment:
you do not need the extra len() check, the for loop will do nothing when the slice is empty.